include/red_black_tree.h

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2019, DSA authors.
+ * All rights reserved.
+ * Authors: Allen
+ */
+
+// 红黑树
+#pragma once
+
+#include <memory>
+#include <queue>
+#include <vector>
+#include <optional>
+#include <iostream>
```

Comment:
I didn't found any usage for line14, and let remove `#include <iostream>`.

Reply:
`using` statement required if you want to use base template's methods and members.